app/src/main/java/org/fossasia/openevent/app/utils/CurrencyUtils.java

```diff
@@ -57,33 +58,43 @@ private CurrencyUtils() {
             .subscribeOn(Schedulers.computation());
     }
 
-    @SuppressWarnings({ "PMD.AvoidCatchingGenericException", "PMD.EmptyCatchBlock" })
-    public static List<String> getCurrencyCodesList() {
-        List<String> currencyCodes = new ArrayList<>();
-        List<String> priorCurrencyCodes = new ArrayList<>();
-
-        priorCurrencyCodes.add(Currency.getInstance(Locale.US).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.UK).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.GERMANY).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.JAPAN).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.CANADA).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.CHINA).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.KOREA).toString());
```

Comment:
These currencies were given priority

Reply:
Priority was given so that the users could select these countries without scrolling much. In the new state, `currency` will automatically be selected in the spinner. So there's no question of scrolling as any item in the middle will be selected as per locale.
From there, if the user wants to change to any one of the priority countries, he will scroll alphabetically and not to the top.

In the loop:
```
for (Locale locale : Locale.getAvailableLocales()) {
            try {
                Currency currency = Currency.getInstance(locale);
                if (currency != null && !currencyCodes.contains(currency.toString())) {
                    currencyCodes.add(currency.toString());
                }
            } catch (Exception e) {
                // No action
            }
        }
```
`priorCurrencyCodes` were being added to the list without checking if they exist:
`priorCurrencyCodes.addAll(currencyCodes);`
I think this is repetitive, and creates a bit of confusion and bad UX.

Reply:
In almost all pickers of currency, you see common ones on the top. Default selected or not. It is very common for someone to live in India and choose USD as default currency

Reply:
I agree that since I live in India, 'INR' will be selected even when 'USD' is needed. But then, to select 'USD' **the most natural action would be to scroll downwards to reach the letter 'U'**. I wouldn't even know/remember that you had placed USD at the top for my convenience.

Google Translate also handles country and language (language analogous to currency in our case) selection similarly:

**Spanish and English was selected by default**

![gtranslate](https://user-images.githubusercontent.com/35009811/38457195-da7cac3c-3aaa-11e8-8686-93c4f5518e6b.png)

Reply:
The list is huge and some of the countries currency are used very often. If we add them to the top and don't remove them from bottom, I guess both problems will be solved